src/ui/webeiddialog.cpp

```diff
@@ -186,9 +174,8 @@ void WebEidDialog::onMultipleCertificatesReady(
     const QUrl& origin, const std::vector<CardCertificateAndPinInfo>& certificateAndPinInfos)
 {
     try {
-        fillOriginAndCertificateList(ui->selectCertificateOriginLabel, ui->selectCertificateInfo,
-                                     origin, certificateAndPinInfos);
-        ui->okButton->setDisabled(true);
+        ui->selectCertificateOriginLabel->setText(fromPunycode(origin));
+        ui->selectCertificateInfo->setCertificateInfo(certificateAndPinInfos);
```

Comment:
Old good [don't repeat yourself](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself), exact same lines are used in 3 places below, so you have to remember changing them everywhere in case of bugs. But I see why you dislike the original function, the four parameters are verbose.

Is there a better way to keep the code DRY?

Reply:
As the pin dialog remove QWidgetList usage then this helper function should have also 6 parameters ja some if defs.
No I don't have right now better solution.

Reply:
Alright, all good then.